packages/protocol/contracts/common/InitializableProxy.sol

```diff
@@ -0,0 +1,19 @@
+pragma solidity ^0.5.13;
+
+import "./Proxy.sol";
+
+/**
+ * @title A Proxy utilizing the Unstructured Storage pattern.
+ * @dev This proxy is intended to be used in conjunction with EIP-1167 minimal proxies.
+ */
+contract InitializableProxy is Proxy {
```

Comment:
Should this extend `InitializableV2` or at least copy the functionality?

Reply:
No, `InitializableV2` uses an unnecessary storage write.

Reply:
IMO, `InitializableV2` should require that the inheriting contract specify an `isInitialized` function which can read another value that is guaranteed to be non-zero if the contract is initialized. I would imagine most contracts have that

Reply:
I would say this could "copy the functionality" in the sense that a constructor could be added that sets the owner to address `0x0`. As long as the deployed InitializableProxy that the Factory points to is owned by a valid address, it could be repointed to a contract that calls `selfdestruct` and break all the EIP-1167 clones.

Reply:
Yeah that's what I meant by copy the functionality, i.e. ensure that the `InitializableProxy` implementation cannot be repointed.

Reply:
Actually why do we even care about the storage write in the constructor? It just happens during deploy no? It doesn't impact all the eip1167 proxies

Reply:
Correct, we don't care about a storage write in the constructor

Reply:
So you are saying you are going to extend `InitializableV2` or copy the functionality?

Reply:
Extending `InitializableV2` doesn't really make sense since we want to keep storage of InitializableProxy unstructured, so as to not mess with the storage of whatever implementation contract is pointed to. I would recommend implementing some sort of analogous behavior as mentioned above (otherwise the off-chain Komenci flow will have to take care of transferring ownership to some invalid address with a separate transaction).

Reply:
> otherwise the off-chain Komenci flow will have to take care of transferring ownership to some invalid address with a separate transaction

This was the plan but I can bake it into the SC

Reply:
Actually I don't think any changes are necessary; the constructor sets the owner to msg.sender, which initializes the contract.

Reply:
So `msg.sender` for the one InitializableProxy deployment we make will be some cLabs controlled address, right? To make the system as trustless as possible we should transfer away to an invalid address so no one can `selfdestruct` the Proxy code. Agree that this can be done off-chain, but feels cleaner to put it here rather than having to remember to add that logic to whatever deployment scripts will initialize this system.

Reply:
> To make the system as trustless as possible we should transfer away to an invalid address so no one can selfdestruct the Proxy code

That's the plan!

How would you integrate this into the SC code?

Reply:
```
constructor() {
    _setOwner(0x0);
}
```

Reply:
Well, that specifically wouldn't work because that would count as "not initialized".

You could of course do `0x1` but I'm not sure I prefer that to just transferring ownership in a separate tx?

Reply:
That's been our approach with `IntializableV2` to set the storage of the implementation contract at constructor time. Why wouldn't we want to do it in this case (with `0x1`)? I don't understand the benefit of doing so in a separate tx?

Reply:
Done